src/lib.rs

```diff
@@ -655,6 +656,23 @@ pub trait Itertools : Iterator {
         TakeWhileRef::new(self, f)
     }
 
+    /// Return an iterator adaptor that iterates over the combinations of
+    /// the elements from an iterator.
+    ///
+    /// Iterator element type is **(Self::Item, Self::Item)**.
+    ///
+    /// ```
+    /// use itertools::Itertools;
+    ///
+    /// let it = (1..5).combinate_pairs();
+    /// assert!(itertools::equal(it, vec![(1, 2), (1, 3), (1, 4), (2, 3), (2, 4), (3, 4)]));
+    /// ```
+    fn combinate_pairs(self) -> CombinatePair<Self> where
+        Self: Sized + Clone
```

Comment:
We should probably put `Self::Item: Clone` here just to give users better help messages.